### DISCUSSION TITLE
[Idea] MetaMask Snap for the XRP Ledger

### DISCUSSION THREAD

**Discussion by intelliot** [XRPLBounties/Proposals#42] [Closed (claimed or cancelled bounty)]
| [View Bounty Proposal Here (Not Merged Yet)](https://github.com/XRPLBounties/Proposals/pull/55) |
| - |

**Comment by TusharPardhe** [comment]
I got a basic idea after reading the proposal and coded a snap which generates wallets and saves the keys(seeds) in metamask that can be used for signing transactions. Each wallet has a unique username and on clicking the username the user can get the seed from metamask. (However, it'll be better if we don't expose the seed else just sign the transaction inside metamask itself)

 It's very very crude at the moment and I want to understand the idea as a whole before proceeding further. 

@intelliot @ckniffen @mvadari Tagging you all for some suggestions.

Link: [XRPL Snap Idea](https://youtu.be/jlNq0ytzASY)

<img width="1450" alt="Screenshot 2022-10-18 at 1 24 02 pm" src="https://user-images.githubusercontent.com/31487192/196429871-953b52e3-62a6-4588-9653-97b584c4aba4.png">

**Reply by mvadari** [reply]
Yes, this looks like a good start

**Comment by JST5000** [comment]
@TusharPardhe Currently the MetaMask Snap Bounty is not ready to be picked up - it was merged by accident before we had budget approval and then reverted, but the previous link in this discussion was misleading because it pointed directly to the merged PR instead of to the repo's codebase. (So it looked like the bounty was merged, even though we reverted it)

Sorry for the confusion!

**Reply by TusharPardhe** [reply]
Hi again @JST5000!

I actually went through all the comments and know that it isn't ready at the moment, but still wanted to learn something new and go through XRPL APIs again just to brush up my knowledge. 

Also, today and tomorrow is Diwali (an Indian festival) so nobody is allowed to sorry's. Only wishes :) 

Happy Diwali!

I will keep a watch on further discussions!

**Reply by JST5000** [reply]
Hahaha, I'm glad :) - I wish you a happy Diwali!

**Comment by golobitch** [comment]
Hi @JST5000,

what is the status of this bounty?
We are planning to release MetaMask Snap next week. Is it possible that it will be approved until then?

**Reply by JST5000** [reply]
We're discussing the bounty in more depth on Monday - but currently still haven't received budgetary approval for it. I don't really have a concrete timeline for when that will be finished unfortunately.

**Reply by JST5000** [reply]
After that discussion, the team decided we have to do some more research before committing budget for this bounty. I'll have a further update in two weeks.

**Comment by golobitch** [comment]
A little bit of an example app preview:
<img width="1014" alt="Screenshot 2022-10-30 at 21 14 54" src="https://user-images.githubusercontent.com/24848451/198899793-25e8c8a2-0ba3-416a-936b-897a521311b3.png">
<img width="794" alt="Screenshot 2022-10-30 at 21 15 23" src="https://user-images.githubusercontent.com/24848451/198899790-263adae7-e7a4-4f1a-9285-4dcdb1492304.png">
